packages/lesswrong/server/callbacks/userCallbacks.tsx

```diff
@@ -79,6 +83,46 @@ getCollectionHooks("Users").editSync.add(function maybeSendVerificationEmail (mo
   }
 });
 
+getCollectionHooks("Users").updateBefore.add(async function updateProfileTagsSubscribesUser(data, {oldDocument, newDocument}: UpdateCallbackProperties<DbUser>) {
+  // check if the user added any tags to their profile
+  const tagIdsAdded = newDocument.profileTagIds?.filter(tagId => !includes(oldDocument.profileTagIds || [], tagId)) || []
+  
+  // if so, then we want to subscribe them to the newly added tags
+  if (tagIdsAdded.length > 0) {
+    const tagsAdded = await Tags.find({_id: {$in: tagIdsAdded}}).fetch()
+    const tagsById = keyBy(tagsAdded, tag => tag._id)
+    
+    let newFrontpageFilterSettings: FilterSettings = newDocument.frontpageFilterSettings || getDefaultFilterSettings()
+    for (let addedTag of tagIdsAdded) {
+      const tagName = tagsById[addedTag].name
+      const existingFilter = newFrontpageFilterSettings.tags.find(tag => tag.tagId === addedTag)
+      // if the user already had a filter for this tag, see if we should update it or leave it alone
+      if (existingFilter) {
+        if (includes([0, 'Default', 'TagDefault'], existingFilter.filterMode)) {
+          newFrontpageFilterSettings = {
+            ...newFrontpageFilterSettings,
+            tags: [
+              ...newFrontpageFilterSettings.tags.filter(tag => tag.tagId !== addedTag),
+              {tagId: addedTag, tagName: tagName, filterMode: 'Subscribed'}
+            ]
+          }
+        }
+      } else {
+        // otherwise, subscribe them to this tag
+        newFrontpageFilterSettings = {
+          ...newFrontpageFilterSettings,
+          tags: [
+            ...newFrontpageFilterSettings.tags,
+            {tagId: addedTag, tagName: tagName, filterMode: 'Subscribed'}
```

Comment:
Seems like `{tagId: addedTag, tagName: tagName, filterMode: 'Subscribed'}` is being used in both branches - would it be appropriate to extract it?  It seems like it'll have the same values, but I'm not clear on whether, if it's changed in the future, we'd generally want to change it in both places or not.